api/v1beta1/azuremachine_types.go

```diff
@@ -86,6 +86,10 @@ type AzureMachineSpec struct {
 	// +optional
 	AdditionalTags Tags `json:"additionalTags,omitempty"`
 
+	// AdditionalCapabilities specifies additional capabilities enabled or disabled on the virtual machine.
+	// +optional
+	AdditionalCapabilities *AdditionalCapabilities `json:"additionalCapabilities,omitempty"`
```

Comment:
So I see that the Azure SDK type spec defines this `AdditionalCapabilities` struct to contain the UltraSSD config... I wonder if we want to expose that to capz users, or simply add `UltraSSDEnabled` to the existing flat configuration property structure in `AzureMachineSpec`.

Thoughts @CecileRobertMichon @mboersma

Reply:
lol I see this was discussed at length in the linked issue

Reply:
Yeah :) we agreed with @CecileRobertMichon on going with this design.